rust/.gitignore

```diff
@@ -1 +1,2 @@
 /target/
+/lib/Cargo.lock
```

Comment:
Lock files are checked in for reproducibility purposes.

See https://github.com/rust-lang/cargo/blob/master/src/doc/src/faq.md#why-have-cargolock-in-version-control but this is indeed a contentious point in the Rust community and there are not yet clear guidelines. Note though that `cargo new` changed behavior to check the lock file, so this is the current default.

Reply:
Okay, I think I had outdated information about keeping `Cargo.lock`. Thank you, I'll refresh my brain's database. 

Anyway, I pushed the commit with dependencies updates only.